pages/settings/script.js

```diff
@@ -1,84 +1,207 @@
 //Show user information
-import { getUserSession, setUserSession } from "../../assets/helpers/storage.js";
+import { getUserSession, setUserSession } from '../../assets/helpers/storage.js'
+import { API_URL } from '../../constants.js'
 
-const user = getUserSession();
+const user = getUserSession()
 
 function showUserInfo() {
-  if (user) {
-    document.getElementById("name").setAttribute("value", user?.name);
-    document.getElementById("email").setAttribute("value", user?.email);
-  } else {
-    myProfile.style.display = "none";
-    deleteAccount.style.display = "none";
-  }
+	if (user) {
+		document.getElementById('name').setAttribute('value', user?.name)
+		document.getElementById('email').setAttribute('value', user?.email)
+	} else {
+		myProfile.style.display = 'none'
+		deleteAccount.style.display = 'none'
+	}
 }
 
-showUserInfo();
+showUserInfo()
 
 // Change user information
-let userName = user?.name;
-let userEmail = user?.email;
-
-saveProfileInfoBtn.addEventListener("click", () => {
-  userName = document.getElementById("name").value;
-  user.name = userName;
-  userEmail = document.getElementById("email").value;
-  user.email = userEmail;
-  setUserSession(user);
-  showProfilePic();
-});
+let userName = user?.name
+let userEmail = user?.email
+let userID = user?.id
+
+saveProfileInfoBtn.addEventListener('click', (e) => {
+	e.preventDefault()
+	userName = document.getElementById('name').value
+	userEmail = document.getElementById('email').value
+	user.name = userName
+	user.email = userEmail
+	setUserSession(user)
```

Comment:
setUserSession(user) should be inside saveUserInfo(), because we only want to update the storage if the request to the endpoint is successful

Reply:
fixed in commit [486adda](https://github.com/whamester/WMDD-4885-Project/pull/15/commits/486adda7af9ef34a8b6a28beef097f09df1c55e5)